src/http_module.cpp

```diff
@@ -6,17 +6,37 @@
 #include "trace_context.hpp"
 #include "batch_exporter.hpp"
 
+#include <fstream>
+#include <sstream>
+
 extern ngx_module_t gHttpModule;
 
 namespace {
 
+grpc::string readFile(const std::string& filePath) {
+    std::ifstream file(filePath);
+
+    if (!file.is_open()) {
+        throw std::runtime_error("Failed to open file: " + filePath);
+    }
+
+    std::stringstream buffer;
+    buffer << file.rdbuf();
+    file.close();
+
+    return grpc::string(buffer.str());
+}
+
 struct OtelCtx {
     TraceContext parent;
     TraceContext current;
 };
 
 struct MainConf {
     ngx_str_t endpoint;
+    ngx_str_t pemRootCerts;
+    ngx_str_t pemCertChain;
+    ngx_str_t pemPrivateKey;
```

Comment:
If I checked well, these two declarations are not needed

Reply:
correct, removed the leftovers